### ISSUE TITLE
Unable to parse flow tuples of unions 

### ISSUE THREAD

**Issue by ayrton** [eslint/eslint#5915]
It's currently impossible to parse [flow tuples of unions](http://flowtype.org/docs/builtins.html#literal-types) in eslint. Is this something you plan to support?

**What version of ESLint are you using?**

1.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint 4.1.6

**What did you do? Please include the actual source code causing the issue.**

```js
type Suit =
  | "Diamonds"
  | "Clubs"
  | "Hearts"
  | "Spades";

function result(suit: Suit): string {
  return 'I had: ' + suit;
}

suit('Clubs');
```

**What did you expect to happen?**

To parse the above JS

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
error  Parsing error: Unexpected token
```

**Comment by platinumazure** [comment]
ESLint doesn't directly support experimental features like FlowType. You should raise this with babel-eslint since you're using that as your parser. Thanks!

**Comment by ilyavolodin** [comment]
Closing as this is an issue for `babel-eslint` or `eslint-plugin-babel`.
